### DISCUSSION TITLE
Add enum class for request parameter

### DISCUSSION THREAD

**Discussion by DharaVaghasiya** [mockoon/mockoon#553] [Help] [answered]
I'm trying to mock api with request parameters:
accountType:	string
                       Enum: checking, savings, loan, investment, crypto_address, unknown, gl, customer_level
description:	 {
                      one:	string
                      two:	string
                      three:	 string
                      four:	string
                      five:	string
                      }

I'm not able to find a way to mock this request parameters,
also would like to request some more complex examples using mockoon and more details in documentation about syntax.

Thank you

**Comment by 255kb** [comment, marked as answer]
Could you please clarify what you are trying to achieve? I am not sure if these are query parameters, URL parameters, or body properties, and what you want to do with them. 
Thank you

**Reply by DharaVaghasiya** [reply]
This is Body parameter for mock service I'm working on. and This request parameter accountType is of string type but value acccepts only from given Enumeration class. and you can ignore discription parameter because I figure out the way for it.

**Reply by 255kb** [reply]
I am still not sure to understand what you want to achieve, but here is a brief overview of what you can do with the entering body in Mockoon: 

- use the entering request's body properties using [`body`](https://mockoon.com/docs/latest/templating/mockoon-request-helpers/#body) or [`bodyRaw`](https://mockoon.com/docs/latest/templating/mockoon-request-helpers/#bodyraw) templating helpers. You can find similar helpers for `queryParam`, `urlParam`, etc. Using these helpers, you can build a dynamic body reusing the request's body:

  ![image](https://user-images.githubusercontent.com/7489814/130634381-1368fe7e-508d-46c2-8ca0-1d0eb9f882df.png)

- serve different responses depending on the body properties by creating [rules](https://mockoon.com/docs/latest/route-responses/dynamic-rules/), here `checking` must be different than 0:
  
  ![image](https://user-images.githubusercontent.com/7489814/130635224-4ba9b99f-3cb1-4463-b745-bcc79f990dbc.png)


Let me know if you need more information.
